src/room/MuteStates.ts

```diff
@@ -74,5 +76,52 @@ export function useMuteStates(): MuteStates {
   const audio = useMuteState(devices.audioInput, () => true);
   const video = useMuteState(devices.videoInput, () => true);
 
+  useEffect(() => {
+    widget?.api.transport.send(ElementWidgetActions.DeviceMute, {
+      audio_enabled: audio.enabled,
+      video_enabled: video.enabled,
+    });
+  }, [audio, video]);
+
+  useEffect(() => {
+    if (widget) {
+      const onMuteStateChangeRequest = (
+        ev: CustomEvent<IWidgetApiRequest>,
+      ): void => {
+        const newState = {
+          audio_enabled: audio.enabled,
+          video_enabled: video.enabled,
+        };
+        if (
+          ev.detail.data.audio_enabled != null &&
+          typeof ev.detail.data.audio_enabled === "boolean"
+        ) {
+          audio.setEnabled?.(ev.detail.data.audio_enabled);
+          newState.audio_enabled = ev.detail.data.audio_enabled;
+        }
+        if (
+          ev.detail.data.video_enabled != null &&
+          typeof ev.detail.data.video_enabled === "boolean"
+        ) {
+          video.setEnabled?.(ev.detail.data.video_enabled);
+          newState.video_enabled = ev.detail.data.video_enabled;
+        }
+        widget!.api.transport.reply(ev.detail, newState);
+      };
+
+      widget.lazyActions.on(
+        ElementWidgetActions.DeviceMute,
+        onMuteStateChangeRequest,
+      );
+
+      return (): void => {
+        widget!.lazyActions.off(
+          ElementWidgetActions.DeviceMute,
+          onMuteStateChangeRequest,
+        );
+      };
+    }
+  }, [audio, video]);
```

Comment:
I might need an explanation on what this is doing: whenever the audio or video mute state changes, we add a new listener for the device mute widget action? The listener presumably also fires (once?) to set the mute states and reply?

Reply:
To my understanding the audio and video elements are not changing instead they mutate. So whenever they are set we connect the listener and when they are unset we remove it.

Reply:
I'm not sure it's that simple. Why not just use a callback and then the problem goes away? Also it will make the code less nested.

Reply:
I updated this. So it is no creating the useCallback and recomputes the on/off when the callback changes.